src/semidiscretization/semidiscretization_parabolic_auxvars.jl

```diff
@@ -0,0 +1,205 @@
+
+# TODO: Taal refactor, Mesh<:AbstractMesh{NDIMS}, Equations<:AbstractEquations{NDIMS} ?
+"""
+    SemidiscretizationParabolicAuxVars
+
+A struct containing everything needed to describe a spatial semidiscretization
+of a parabolic conservation law.
+"""
+struct SemidiscretizationParabolicAuxVars{Mesh, Equations, InitialCondition, BoundaryConditions,
+                                          SourceTerms, Solver, Cache} <: AbstractSemidiscretization
+
+  mesh::Mesh
+  equations::Equations
+
+  # This guy is a bit messy since we abuse it as some kind of "exact solution"
+  # although this doesn't really exist...
+  initial_condition::InitialCondition
+
+  boundary_conditions::BoundaryConditions
+  source_terms::SourceTerms
+  solver::Solver
+  cache::Cache
+  performance_counter::PerformanceCounter
+
+  function SemidiscretizationParabolicAuxVars{Mesh, Equations, InitialCondition, BoundaryConditions,
+                                              SourceTerms, Solver, Cache}(
+      mesh::Mesh, equations::Equations,
+      initial_condition::InitialCondition, boundary_conditions::BoundaryConditions,
+      source_terms::SourceTerms,
+      solver::Solver, cache::Cache) where {Mesh, Equations, InitialCondition, BoundaryConditions, SourceTerms, Solver, Cache}
+    @assert ndims(mesh) == ndims(equations)
+
+    performance_counter = PerformanceCounter()
+
+    new(mesh, equations, initial_condition, boundary_conditions, source_terms, solver, cache, performance_counter)
+  end
+end
+
+"""
+    SemidiscretizationParabolicAuxVars(mesh, equations, initial_condition, solver;
+                                 source_terms=nothing,
+                                 boundary_conditions=boundary_condition_periodic)
+
+Construct a semidiscretization of a parabolic PDE.
+"""
+function SemidiscretizationParabolicAuxVars(mesh, equations, initial_condition, solver;
+                                            source_terms=nothing,
+                                            boundary_conditions=boundary_condition_periodic,
+                                            RealT=real(solver))
+
+  cache = create_cache(mesh, equations, solver, RealT)
+  _boundary_conditions = digest_boundary_conditions(boundary_conditions)
+
+  solver_auxvars = create_solver_auxvars(solver)
```

Comment:
We will need to have the possibility to use different numerical fluxes for the first and second step (computation of auxiliary variables and doing stuff with them) for methods different from BR1.

Reply:
Absolutely. That's one of the reasons I wanted the parabolic terms to be handled by a separate solver. I originally planned to postpone this to a second PR, but I think we can also put it in here. Maybe we can discuss this next Monday?